test/test_quantized.py

```diff
@@ -1089,420 +1089,422 @@ def test_qlinear(self, batch_size, input_channels, output_channels,
         else:
-            W = torch.from_numpy(_dequantize(
-                W_q0, W_scales[0], W_zps[0])).to(dtype=torch.float)
-            W_q = torch.quantize_per_tensor(W, scale=W_scales[0], zero_point=(
-                W_zps[0].astype(int).item()), dtype=torch.qint8)
-            b = torch.from_numpy(_dequantize(
-                b_q0, X_scale * (W_scales[0].item()), 0)).to(dtype=torch.float) if use_bias else None
-            b_q = torch.quantize_per_tensor(
-                b, scale=X_scale * (W_scales[0].item()), zero_point=0, dtype=torch.qint32) if use_bias else None
-
-        # Compare X_scale * W_scale * input_channels * X_value_max * W_value_max with
-        # Y_scale * 255 (max for uint8).
-        Y_scale = 125.1234
-        Y_zp = 5
-
-        # Weight prepacking operator for quantized Linear
-        float_bias = b if use_bias else None
-        W_prepack = qlinear_prepack(W_q, float_bias)
-
-        if use_multi_dim_input:
-            X_q = X_q.view(3, int(batch_size / 3), input_channels)
-
-        # Quantized Linear operator with prepacked weight
-        Y_q = qlinear(X_q, W_prepack, Y_scale, Y_zp)
-
-        if not use_channelwise:
-            # Test the per-tensor quantization only
-            # Reference quantized Linear operator
-            Y_q_ref = qlinear_ref(X_q0, X_scale, X_zp, W_q0,
-                                  W_scales[0], W_zps[0], b_q0, Y_scale, Y_zp)
-            if use_relu:
-                Y_q_ref[Y_q_ref < Y_zp] = Y_zp
-            if use_multi_dim_input:
-                Y_q_ref = np.reshape(
-                    Y_q_ref, (3, int(batch_size / 3), output_channels))
-
-            # Assert equal
             np.testing.assert_equal(Y_q_ref, Y_q.int_repr().numpy())
-
-        # Test both per-tensor and per-channel quantization
-        # Reference quantized result from PyTorch Linear operator
-        W_fp32 = W_q.dequantize().to(dtype=torch.float)
-        X_fp32 = X_q.dequantize().to(dtype=torch.float)
-        b_fp32 = b_q.dequantize().to(dtype=torch.float) if use_bias else None
-        Y_fp32_ref = F.linear(X_fp32, W_fp32, b_fp32)
-        if use_relu:
-            Y_fp32_ref[Y_fp32_ref < 0.0] = 0.0
-        Y_q_ref2 = torch.quantize_per_tensor(
-            Y_fp32_ref, Y_scale, Y_zp, torch.quint8)
-        # Assert equal
+    # Test both per-tensor and per-channel quantization
+    # Reference quantized result from PyTorch Linear operator
+    W_fp32 = W_q.dequantize().to(dtype=torch.float)
+    X_fp32 = X_q.dequantize().to(dtype=torch.float)
+    b_fp32 = b_q.dequantize().to(dtype=torch.float) if use_bias else None
+    Y_fp32_ref = F.linear(X_fp32, W_fp32, b_fp32)
+    if use_relu:
+        Y_fp32_ref[Y_fp32_ref < 0.0] = 0.0
+    Y_q_ref2 = torch.quantize_per_tensor(
+        Y_fp32_ref, Y_scale, Y_zp, torch.quint8)
+    # Assert equal
+    if qengine == 'qnnpack':
+        np.testing.assert_array_almost_equal(
+            Y_q_ref2.int_repr().numpy(), Y_q.int_repr().numpy(), decimal=0)
```

Comment:
maybe just set `decimal` in the beginning to 0 or None (or whatever default value) and use it everywhere?